source/common/upstream/load_balancer_impl.h

```diff
@@ -74,13 +76,33 @@ class LoadBalancerBase {
   /**
    * Regenerate locality aware routing structures for fast decisions on upstream locality selection.
    */
-  void regenerateLocalityRoutingStructures();
-
-  const HostSet* local_host_set_;
-  uint64_t local_percent_to_route_{};
-  LocalityRoutingState locality_routing_state_{LocalityRoutingState::NoLocalityRouting};
-  std::vector<uint64_t> residual_capacity_;
-  Common::CallbackHandle* local_host_set_member_update_cb_handle_{};
+  void regenerateLocalityRoutingStructures(uint32_t priority);
+
+  uint32_t best_available_priority() { return best_available_host_set_->priority(); }
+
+  HostSet& local_host_set() { return *local_priority_set_->hostSetsPerPriority()[0]; }
+
+  // The priority-ordered set of hosts to use for load balancing.
+  const PrioritySet& priority_set_;
+  // The lowest priority host set from priority_set_ with healthy hosts, or the
+  // zero-priority host set if all host sets are fully unhealthy.
+  const HostSet* best_available_host_set_;
+
+  // The set of local Envoy instances which are load balancing across priority_set_
+  const PrioritySet* local_priority_set_;
+
+  struct PerPriorityState {
+    uint64_t local_percent_to_route_{};
+    LocalityRoutingState locality_routing_state_{LocalityRoutingState::NoLocalityRouting};
+    std::vector<uint64_t> residual_capacity_;
```

Comment:
I know this was an issue in the original, but I think it would help readability to explain what these fields purpose in life is.